File: Minh_MyOrder/app/src/main/java/com/example/minh_myorder/MainActivity.java
package com.example.minh_myorder;

import android.os.Bundle;
import android.util.Log;
import android.view.View;

import androidx.appcompat.app.AppCompatActivity;

import com.example.minh_myorder.databinding.ActivityMainBinding;

public class MainActivity extends AppCompatActivity {

    private ActivityMainBinding binding;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        binding = ActivityMainBinding.inflate(getLayoutInflater());
        setContentView(binding.getRoot());
    }

    public void onRadioButtonClicked(View v){
        switch(v.getId()){
            case R.id.rb_espresso:
                this.binding.tvCoffeeStylePrompt.setVisibility(View.INVISIBLE);
                this.binding.rgCoffeeStyle.setVisibility(View.INVISIBLE);
                break;
            case R.id.rb_coffee:
                Log.d("Coffee App", "onRadioButtonClicked: clicked!");
                this.binding.tvCoffeeStylePrompt.setVisibility(View.VISIBLE);
                this.binding.rgCoffeeStyle.setVisibility(View.VISIBLE);
                break;
            case R.id.rb_cappuccino:
                this.binding.tvCoffeeStylePrompt.setVisibility(View.INVISIBLE);
                this.binding.rgCoffeeStyle.setVisibility(View.INVISIBLE);
                break;
        }
    }
}

File: TipCalculator/app/src/main/java/com/example/tipcalculator/MainActivity.kt
package com.example.tipcalculator

import android.app.Application
import android.os.Bundle
import android.widget.Toast
import androidx.appcompat.app.AppCompatActivity
import com.example.tipcalculator.databinding.ActivityMainBinding
import java.text.NumberFormat


class MainActivity : AppCompatActivity() {

    private lateinit var binding: ActivityMainBinding


    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        binding = ActivityMainBinding.inflate(layoutInflater)
        setContentView(binding.root)

        binding.calculateButton.setOnClickListener {
           this.calculateTip()
        }

    }

    private fun calculateTip() {
        if (binding.costOfService.text.toString().trim().isEmpty()){
            Toast.makeText(applicationContext, "No service cost was entered", Toast.LENGTH_SHORT).show()
        }else {
            val serviceCost = binding.costOfService.text.toString().toDouble()

            val tipPercentage = when (binding.tipOptions.checkedRadioButtonId) {
                R.id.option_twenty_percent -> 0.20
                R.id.option_eighteen_percent -> 0.18
                else -> 0.15
            }
            var tip = serviceCost * tipPercentage
            if (binding.roundUpTip.isChecked) {
                tip = kotlin.math.ceil(tip)
            }
            val formattedTip = NumberFormat.getCurrencyInstance().format(tip)
            binding.tipAmount.text = getString(R.string.tip_amount, formattedTip)
        }
   }

}